src/data/actions/utils.js

```diff
@@ -25,39 +26,65 @@ const sortAlphaAsc = (gradeRowA, gradeRowB) => {
 
 const sortNumerically = (colKey, direction) => {
   function sortNumAsc(gradeRowA, gradeRowB) {
-    if (gradeRowA[colKey] < gradeRowB[colKey]) {
+    let a;
+    let b;
+    if (colKey !== 'total') {
+      a = gradeRowA.section_breakdown.find(x => x.label === colKey).percent;
+      b = gradeRowB.section_breakdown.find(x => x.label === colKey).percent;
+    } else {
+      a = gradeRowA.percent;
+      b = gradeRowB.percent;
+    }
+
+    if (a < b) {
       return -1;
     }
-    if (gradeRowA[colKey] > gradeRowB[colKey]) {
+    if (a > b) {
       return 1;
     }
     return 0;
   }
 
   function sortNumDesc(gradeRowA, gradeRowB) {
-    if (gradeRowA[colKey] < gradeRowB[colKey]) {
+    let a;
```

Comment:
Everything from here to the `if (a < b)` statement is exactly the same as the lines above.  It's probably worth extracting this to a shared function.